lms/djangoapps/mobile_api/friends/views.py

```diff
@@ -0,0 +1,96 @@
+"""
+Views for friends info API
+"""
+
+from rest_framework import generics, permissions, status
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
+from rest_framework.response import Response
+from openedx.core.djangoapps.user_api.api.profile import preference_info
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+from social.apps.django_app.default.models import UserSocialAuth
+from student.models import CourseEnrollment
+import serializers
+import urllib2
+import json
+import facebook
+
+# TODO: change this to final config
+from ..settings import FB_SETTINGS
+_FACEBOOK_API_VERSION = FB_SETTINGS['_FACEBOOK_API_VERSION']
+
+
+class FriendsInCourse(generics.ListAPIView):
+    """
+    **Use Case**
+
+        API endpoint that returns all the users friends that are in the course specified.
+        Note that only friends that allow their courses to be shared will be included.
+
+    **Example request**:
+
+        GET /api/mobile/v0.5/friends/course/<course_id>
+
+        where course_id is in the form of /edX/DemoX/Demo_Course
+
+    **Response Values**
+
+        {   "friends":
+                [{
+                    "name": "test",
+                    "id": "12345",
+                    "edX_id": "678910"
+                },
+                ...
+                ]
+            }
+    """
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.FriendsInCourseSerializer
+
+    def list(self, request, *args, **kwargs):
```

Comment:
The logic in this method is very similar to the corresponding method for the `CoursesWithFriends` endpoint.  Let's consolidate these 2 endpoints.  The only difference being that `FriendsInCourse` will filter for a specific course, while `CoursesWithFriends` will filter for any matching course.